spynnaker/pyNN/models/spike_source/spike_source_poisson_machine_vertex.py

```diff
@@ -246,20 +246,18 @@ def __init__(
             label, app_vertex=app_vertex, vertex_slice=vertex_slice)
         self.__is_recording = is_recording
         self.__sdram = sdram
-        self.__sdram_partition: Optional[AbstractSDRAMPartition] = None
+        self.__sdram_partition: Optional[
+            SourceSegmentedSDRAMMachinePartition] = None
         self.__rate_changed = True
 
     @property
     def _pop_vertex(self) -> SpikeSourcePoissonVertex:
         return cast('SpikeSourcePoissonVertex', self.app_vertex)
 
+    @overrides(SendsSynapticInputsOverSDRAM.set_sdram_partition)
```

Comment:
docs in abstract says it should never be called twice. This allows that

Reply:
Fixed